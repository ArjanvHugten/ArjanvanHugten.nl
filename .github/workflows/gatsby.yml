name: Gatsby CI

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - name: Checkout
      uses: actions/checkout@v2.0.0
    - name: Build Gatsby Site
      uses: jzweifel/gatsby-cli-github-action@master
      with:
        gatsby-arg: build
    - name: Run Jest
      uses: stefanoeb/jest-action@1.0.3
      with:
        jestArgs: --silent
