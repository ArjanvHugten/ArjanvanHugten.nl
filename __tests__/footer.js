import React from "react"
import renderer from "react-test-renderer"
import { render, screen } from '@testing-library/react'
import { useStaticQuery } from "gatsby"

import Footer from "../src/components/layout/footer"

beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      "footerData": {
          "edges": [{
              "node": {
                  "frontmatter": {
                      "text": "This is the footer" 
                  }
              }
          }]
      }
    }))
})  

describe("Footer", () => {
  it("Renders correctly", () => {
    const testRenderer  = renderer.create(<Footer />)

    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it("Checks footer text setted correctly", () => {
    render(<Footer />)
    expect(screen.getByText(/Copyright/i)).toHaveTextContent(`Copyright Â© ${new Date().getFullYear()} -`)
    expect(screen.getByText("This is the footer")).toBeInTheDocument()
  })
})