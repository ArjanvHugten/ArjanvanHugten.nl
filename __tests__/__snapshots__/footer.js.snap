// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders correctly 1`] = `
<footer>
  <div
    className="container has-text-centered"
  >
    <hr />
    <p>
      Copyright Â© 
      2020
       - 
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is the footer",
          }
        }
      />
    </p>
  </div>
</footer>
`;
